helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -200,6 +201,22 @@ private void joinCluster() {
     }
   }
 
+  private String ConstructDomainField(String faultDomain) {
+    Boolean topologyEnabled = _configAccessor.getClusterConfig(_clusterName).isTopologyAwareEnabled();
```

Comment:
Do we need this? If we know the environment is Azure, then we can directly get the topology from AzureConstants instead of reading from ClusterConfig.

Reply:
In this logic, we do not parse the cloud provider directly. That logic is in the processor. I was thinking that if we read from clusterConfig, it'll be generic, and independent of cloud provider. thoughts?

Reply:
Discussed offline. We'll move the logic to cloud instance information processor and not to depend on cluster config.